lib/ClangImporter/ClangImporter.cpp

```diff
@@ -5652,14 +5652,20 @@ ClangImporter::getCXXFunctionTemplateSpecialization(SubstitutionMap subst,
 
 bool ClangImporter::isCXXMethodMutating(const clang::CXXMethodDecl *method) {
   return isa<clang::CXXConstructorDecl>(method) || !method->isConst() ||
-         method->getParent()->hasMutableFields() ||
-         (method->hasAttrs() &&
-          llvm::any_of(method->getAttrs(), [](clang::Attr *a) {
-            if (auto swiftAttr = dyn_cast<clang::SwiftAttrAttr>(a)) {
-              return swiftAttr->getAttribute() == "mutating";
-            }
-            return false;
-          }));
+         (method->getParent()->hasMutableFields() &&
+          !isAnnotatedWith(method, "nonmutating")) ||
+         isAnnotatedWith(method, "mutating");
+}
+
+bool ClangImporter::isAnnotatedWith(const clang::CXXMethodDecl *method,
```

Comment:
This will be very helpful. I suspect there are a few places we could use this already :)